sources/packages/web/src/composables/useFormioComponentLoader.ts

```diff
@@ -9,21 +13,55 @@ import { useFormioUtils } from ".";
  */
 export function useFormioComponentLoader() {
   const formioUtils = useFormioUtils();
-  // Get offering date of the selected offering and set to the hidden field (selectedOfferingDate) in formio.
-  const loadSelectedOfferingDate = async (
+
+  /**
+   * Get details of given offering and populate form.io
+   * hidden fields.
+   * @param form form to be updated with hidden field values.
+   * @param offeringId offering.
+   * @param offeringFieldIds hidden field names in form.io.
+   * @param options API options to get offering details.
+   */
+  const loadSelectedOfferingDetails = async (
```

Comment:
This method seems a bit harder to follow, IMO, tt seems that we are trying to merge two methods into one.
For instance,
- for one situation it will be accessed from the student's perspective, using one API call and setting start and end dates.
- for the second situation it will be accessed by the institution, using a second API and setting only the end date.
Also, the `options` is used to define if the institution API or the student API must be invoked.
I am struggling to see the benefit of having both logics merged.